### ISSUE TITLE
Translation performance: _csv_to_array() fails at string with embedded quotes

### ISSUE THREAD

**Issue by rwetzlmayr** [SocialEngine/phpv4-issues#140]
Steps to reproduce:
1. Add a new language entry with an embedded double quote via Language Manager, e.g. `foo "bar" baz` as the source string - the target string does not matter. This will add `"foo ""bar"" baz";"foo ""bar"" baz"` at the end of `languages/xx/custom.csv`.
2. Check 'Translation Performance: Convert Language Pack CSV files to a PHP Array?' in **Settings > Performance and Caching**
3. Hit **Save Changes**
4. Look at the corresponding array entry in the converted PHP file `langauges/xx/xx.php`to find that its key has been truncated to `'"bar" baz' => 'foo "bar" baz'`. Thus the translated string will never be found at runtime.

The culprit is contained in the `\Core_Controller_Action_Admin::_csv_to_array()` function around line 150ff:

```
            // Make Sure Incoming phrase starts with Double Quotes (")
            $start = strpos( $phrase[0], '"');
            if( $start != 0 ) {
              $phrase[0] =  substr( $phrase[0], $start );
              }
```

I don't think that incoming phrases will ever start with double quotes as double quotes are set as a delimiter in the preceding call to `fgetcsv( $data, 0, ';', '"', '\#' )`, so this whole part of `_csv_to_array()`looks rather pointless and potentially buggy.

**Comment by Anshuse** [comment]
Hi @rwetzlmayr ,

We're able to reproduce this issue at our end and have reported it in our internal bug tracker for our development team to resolve this in near future release.

Best Regards,
Anshu

**Comment by alimoosavi15** [comment]
This is also failing CSV to Array performance feature.

Phrase:
`The ""Reply-To"" address for emails`
Converts to:
`"Reply-To" address for emails`

Phrase:
`I have read and agree to the <a target=""_blank"" href=""%s/help/terms"">terms of service</a>.`
Converts to:
`"_blank" href="%s/help/terms">terms of service</a>.`

"Online Members" list.
Converted entries are getting cut off thus not being rendered in front-end.

**Comment by sgreiner** [comment]
Here is a patch, which fixes the weird transaltions of the PHP translation arrays for me.

File: application/modules/Core/Controller/Action/Admin.php
Patch: (you can ignore the first part with the escapeshellarg if you already have v4.8.11)
```
Index: Admin.php
===================================================================
--- Admin.php	(revision 1633)
+++ Admin.php	(working copy)
@@ -127,8 +127,8 @@
     fclose($fp);
 
     if (function_exists('exec')) {
-    exec('php -l ' . escapeshellarg($fp), $output, $ret);
-      return $ret == 0;
+      exec('php -l ' . escapeshellarg($php_array_file), $output, $ret);
+      return $ret == 0 || empty($output);
     }
 
     return true;
@@ -138,45 +138,35 @@
       $file_array = array();
       if( ( $data = fopen($csv_file, 'r')) !== FALSE) {
           // ignore first characters of file until double quotes are found (")
-          while( ( $phrase = fgetcsv( $data, 0, ';', '"', '\#' ) ) !== FALSE ) {
+        while (($phrase = fgetcsv($data, 0, ';', '"')) !== false) {
 
-            // Make Sure Incoming phrase starts with Double Quotes (")
-            $start = strpos( $phrase[0], '"');
-            if( $start != 0 ) {
-              $phrase[0] =  substr( $phrase[0], $start );
-              }
+          // Make sure incoming phrase is not commented out
+          if (strpos($phrase[0], '#') === 0) {
+            continue;
+          };
 
-            // If First and Last charachters are " double quotes, remove them
-            if( strpos( $phrase[0], '"') == 0 && strrpos($phrase[0] , '"') == (strlen( $phrase[0] ) - 1 ) ) {
-              $phrase[0] = substr($phrase[0], 1,  -1);
+          $phrase_count = count($phrase);
+          if ($phrase_count == 2) {
+            // Add Singular Phrases
+            $file_array[$phrase[0]] = $phrase[1];
+          } elseif ($phrase_count > 2) {
+            // Add Pluralized Phrases
+            $plural_array = array();
+            for ($c = 1; $c < $phrase_count; $c++) {
+              $plural_array[] = $phrase[$c];
             }
-
-            $phrase_count = count( $phrase );
-            if( $phrase_count == 2 ) {
-              // Add Singular Phrases
-              $file_array[ $phrase[0] ] = $phrase[1];
-            }
-            elseif( $phrase_count > 2 ) {
-              // Add Pluralized Phrases
-              $plural_array = array();
-              for($c = 1; $c < $phrase_count; $c++ ) {
-                $plural_array[] = $phrase[$c];
-              }
-              $file_array[ $phrase[0] ] = $plural_array;
-            }
-
-            else{
-              // Do nothing for Blank Phrases
-            }
-
+            $file_array[$phrase[0]] = $plural_array;
+          } else {
+            // Do nothing for Blank Phrases
+            continue;
           }
+        }
+      }
 
-        }
-        
-      return $file_array;   
+      return $file_array;
 }
-  
-protected function _check_phrase( $phrase ) {
+
+  protected function _check_phrase( $phrase ) {
   // Check for Place Holders (pregreplace $ with \$)  
   $phrase = preg_replace('/\$/', '\\\$', $phrase);
   
@@ -200,4 +190,4 @@
   
   return $phrase;
   }
-}
\ No newline at end of file
+}

```

**Comment by Khushboo016** [comment]
Hi @sgreiner , We have forwarded your patch for fixing the issue to our development team for internal team discussion.

Thanks!

**Comment by DonnaScriptTechs** [comment]
This is fixed for our 4.9.3 version when it is released.
